### PR TITLE
Fix bundled warnings

### PR DESCRIPTION
**Pull request by masterleinad**
After #3897, `clang` gives the warning
```
 anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
```
in the bundled `boost` and `muParser`. The changes to `boost` coincide with the (relevant part of) https://github.com/boostorg/random/pull/32.

### CODE REVIEW

File: bundled/boost-1.62.0/include/boost/random/binomial_distribution.hpp [hunk 1]
```diff
@@ -278,18 +278,18 @@ class binomial_distribution {
         m = static_cast<IntType>((t+1)*p);
 
         if(use_inversion()) {
-            q_n = pow((1 - p), static_cast<RealType>(t));
+            _u.q_n = pow((1 - p), static_cast<RealType>(t));
```
Comment (bangerth) [inline bundled/boost-1.62.0/include/boost/random/binomial_distribution.hpp]: Hm. Are these changes they've also made upstream in BOOST?

Comment (bangerth) [reply bundled/boost-1.62.0/include/boost/random/binomial_distribution.hpp]: Ah, sorry. Should have read the PR text above.

Comment (masterleinad) [reply bundled/boost-1.62.0/include/boost/random/binomial_distribution.hpp]: Yes, the PR I referenced is merged.

**Pull request merged**